### PR TITLE
[feat]: create file based on full path

### PR DESCRIPTION
**Pull request by lukePeavey**
For #2

This still needs some additional work. But I thought I would start a PR so could take a look at what I have so far.

### Changes
The first modal (choose folder) is unchanged
The second modal (file name) now accepts a **full file path**.   If the input is just a file name, the plugin will work exactly as before. If the input is a path, any directories in the path will be created if they do not already exist.  

https://user-images.githubusercontent.com/8286271/163267550-3699ec7d-27e3-4ea4-9bba-a0d9afeef44e.mp4

### TODO: 

**Windows support**

Need to test on Windows to make sure it supports "foo\bar\file name" 

**Mobile support** 

On mobile, the File Explorer does not update correctly when the new file path creates multiple directories.  The file gets created in the correct location and is accessible via switcher.  But file explorer only adds the outer directory. You have to reload the app to get it to update.  

Example: 

Input "foo/bar/file name" 
Result: File Explorer adds an empty  "foo" directory

So far, I haven't been able to figure out whats causing this. Let me know if you have any thoughts.

### CODE REVIEW

**Comment by vanadium23** [comment]
Looks good!
I will test on my windows desktop to check if there any issues.

File: .prettierrc.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (vanadium23) [inline .prettierrc.json:1]: Thanks for prettier config. :+1:

Comment (lukePeavey) [reply .prettierrc.json:1]: No prob!  Meant to ask you if you wanted to use Prettier....

Comment (lukePeavey) [reply .prettierrc.json:1]: If you want I can put that in a separate PR so we don't have a bunch of formatting changes mixed in here.

**Comment by lukePeavey** [comment]
I found a workaround for the file explorer issue in the mobile IOS app.   Instead of creating the directory recursively with `vault.createFolder`,  you have to create each directory in the path one at time.  I've tested on iPad and mac, seems to be working on both now. I'll push the changes today.

Also, wanted to suggest a small formatting changes: 
Are you OK with splitting the `modal.ts` file into two separate files: `ChooseFolderModal.ts` and `CreateNoteModal.ts`?

**Comment by vanadium23** [comment]
Yes, let's split file.

File: .prettierrc.json (hunk 1 above)
Comment (vanadium23) [reply .prettierrc.json:1]: No, there are small number of changed line.

File: src/utils.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  /** The file name without extension */
+  name: string;
+}
+
+export const path = {
+  /**
+   * Parses a path string into directory and file name
+   * Supports '/foo/bar/file' or '\foo\bar\file'
+   */
+  parse(path: string): ParsedPath {
```
Comment (vanadium23) [inline src/utils.ts:15]: May be we can use builtin node.js functions?
https://nodejs.org/api/path.html#pathdirnamepath

Comment (lukePeavey) [reply src/utils.ts]: I tried that initially but it only worked on desktop.

Comment (lukePeavey) [reply src/utils.ts]: I was getting an error in the IOS app when I tried to use built in node packages.

Comment (lukePeavey) [reply src/utils.ts]:
```js 
import  * as path from 'path'

...

path.join("strings"
```

Results in an error on the IOS app.  Not sure why.

Comment (vanadium23) [reply src/utils.ts]: Okay :(

**Comment by vanadium23** [comment]
Everything works as expected on windows and android.
I've tested on windows either with `/` and `\`. Works both way 👍

**Comment by vanadium23** [comment]
Looks good to merge. @lukePeavey are you done all you want? Can I merge and release?

**Comment by lukePeavey** [comment]
Great! if you're good with everything, you  can go ahead and merge

**Comment by lukePeavey** [comment]
Pushed one more commit, just prettier to dev dependencies.

**Pull request merged**